sdk/nodejs/runtime/resource.ts

```diff
@@ -227,39 +233,51 @@ async function prepareResource(label: string, res: Resource, custom: boolean,
     /** IMPORTANT!  We should never await prior to this line, otherwise the Resource will be partly uninitialized. */
 
     // Before we can proceed, all our dependencies must be finished.
-    const explicitDependencies: URN[] = await gatherExplicitDependencies(opts.dependsOn);
+    const explicitDependentResources = new Set(await getExplicitDependentResources(opts.dependsOn));
 
     // Serialize out all our props to their final values.  In doing so, we'll also collect all
     // the Resources pointed to by any Dependency objects we encounter, adding them to 'propertyDependencies'.
-    const propertyDependencies: Record<string, Resource[]> = {};
-    const serializedProps = await serializeResourceProperties(label, props, propertyDependencies);
-
-    let parentURN: URN | undefined;
-    if (opts.parent) {
-        parentURN = await opts.parent.urn.promise();
-    } else {
-        // If no parent was provided, parent to the root resource.
-        parentURN = await getRootResource();
-    }
+    const [serializedProps, propertyToDependentResources] = await serializeResourceProperties(label, props);
+
+    // Wait for the parent to complete.
+    // If no parent was provided, parent to the root resource.
+    const parentURN = opts.parent
+        ? await opts.parent.urn.promise()
+        : await getRootResource();
+
+    // // Also, wait for any of our previous siblings to complete.
+    // // NOTE(cyrus): Do we really want to do this?  This is nice so that consumers don't have to
+    // // pass siblings in as dependencies of other children.  However, it limits our ability to
+    // // parallelize child resources of a component resource.
+    // if (opts.parent) {
+    //     await getTransitivelyDependendentURNs(opts.parent.__childResources, res);
+    // }
```

Comment:
@swgillespie @lukehoban curious what you think about this.

Reply:
I think that we should leave this out. It feels too cute to me.